lib/ansible/modules/system/gconftool2.py

```diff
@@ -36,146 +38,233 @@
-    def __init__(self, ansible, key, value_type, value,
-                 direct=False, config_source=""):
-        self.ansible = ansible
-        self.key = key
-        self.value_type = value_type
-        self.value = value
-        self.config_source = config_source
-        self.direct = direct
-
-    def value_already_set(self):
-        return False
-
-    def call(self, call_type, fail_onerr=True):
-        """ Helper function to perform gconftool-2 operations """
-        config_source = ''
-        direct = ''
-        changed = False
-        out = ''
-
-        # If the configuration source is different from the default, create
-        # the argument
-        if self.config_source is not None and len(self.config_source) > 0:
-            config_source = "--config-source " + self.config_source
-
-        # If direct is true, create the argument
-        if self.direct:
-            direct = "--direct"
-
-        # Execute the call
-        cmd = "gconftool-2 "
-        try:
-            # If the call is "get", then we don't need as many parameters and
-            # we can ignore some
-            if call_type == 'get':
-                cmd += "--get {0}".format(self.key)
-            # Otherwise, we will use all relevant parameters
-            elif call_type == 'set':
-                cmd += "{0} {1} --type {2} --{3} {4} \"{5}\"".format(direct,
-                                                                     config_source,
-                                                                     self.value_type,
-                                                                     call_type,
-                                                                     self.key,
-                                                                     self.value)
-            elif call_type == 'unset':
-                cmd += "--unset {0}".format(self.key)
-
-            # Start external command
-            process = Popen([cmd], stdout=PIPE, stderr=PIPE, shell=True)
-
-            # In either case, we will capture the output
-            out = process.stdout.read()
-            err = process.stderr.read()
-
-            if len(err) > 0:
-                if fail_onerr:
-                    self.ansible.fail_json(msg='gconftool-2 failed with '
-                                               'error: %s' % (str(err)))
-            else:
-                changed = True
-
-        except OSError:
-            exception = get_exception()
-            self.ansible.fail_json(msg='gconftool-2 failed with exception: '
-                                       '%s' % exception)
-        return changed, out.rstrip()
+from ansible.module_utils.basic import AnsibleModule
+
+
+class GConfTool2(object):
+    """
+    Small wrapper class around the gconftool-2 binary that can be used for
+    manipulating a single preference in GConf repository.
+    """
+
+    def __init__(self, module, direct=False, config_source=None, check_mode=False):
```

Comment:
Seems to be a mixed bag. `hg`, `bzr` use same approach of getting params out in `main()`, `service` takes in just the module object. Looks like 215 with just module in `__init__` vs 100 with other params as well, based on some quick grepping:

```
grep -r __init__ lib/ansible/modules/  | grep module | grep  'module)' | wc -l
grep -r __init__ lib/ansible/modules/  | grep module | grep -v 'module)' | wc -l
```